### PR TITLE
Fix permissions table interface for mobile

### PR DESCRIPTION
**Pull request by paescuj**
## Scope

Ensure the permissions table under policies is displayed correctly on mobile devices:
- Drop stickiness of header (as discussed in #22952)
- Make table scrollable
- Prevent wrapping of "All / None" buttons

https://github.com/directus/directus/assets/5363448/f582a40c-0274-49df-9a10-70b0a3bce43c

https://github.com/directus/directus/assets/5363448/7fef1849-154b-4f3c-9d1f-bb72287bae98

## Potential Risks / Drawbacks

- Header not sticky anymore, though acceptable as described in https://github.com/directus/directus/pull/22952#issuecomment-2222449006

## Review Notes / Questions

- Down arrow of button in footer moves along with the scroll - this will addressed separately in the button component itself

### CODE REVIEW

File: app/src/interfaces/_system/system-permissions/system-permissions.vue [hunk 1]
```diff
@@ -570,117 +569,96 @@ function useGroupedPermissions() {
...
-				{{ t('reset_system_permissions_copy') }}
-			</v-card-title>
-			<v-card-actions>
-				<v-button secondary @click="resetActive = false">{{ t('cancel') }}</v-button>
-				<v-button @click="resetSystemPermissions(resetActive === 'recommended')">
-					{{ t('reset') }}
-				</v-button>
-			</v-card-actions>
-		</v-card>
-	</v-dialog>
```
Comment (hanneskuettner) [inline app/src/interfaces/_system/system-permissions/system-permissions.vue:676]: Why was this removed?

File: app/src/interfaces/_system/system-permissions/system-permissions.vue [hunk 2]
```diff
@@ -570,117 +569,96 @@ function useGroupedPermissions() {
...
+						:collection="group.collection"
+						@edit-item="editItem(group.collection.collection, $event)"
+						@remove-row="removeCollection(group.collection.collection)"
+						@set-full-access-all="setFullAccessAll(group.collection.collection)"
+						@set-no-access-all="setNoAccessAll(group.collection.collection)"
+						@set-full-access="setFullAccess(group.collection.collection, $event)"
+						@set-no-access="setNoAccess(group.collection.collection, $event)"
+					/>
+
+					<tr v-if="regularPermissions.length > 0 && systemPermissions.length > 0">
```
Comment (hanneskuettner) [inline app/src/interfaces/_system/system-permissions/system-permissions.vue:601]: If we are doing `<tfoot>`, what do you think about moving this row into it as well, since it always is the last row (before the add button) and does not have the same semantic meaning as the body does?

File: app/src/interfaces/_system/system-permissions/system-permissions.vue (hunk 1 above)
Comment (paescuj) [reply app/src/interfaces/_system/system-permissions/system-permissions.vue:676]: dontknow

File: app/src/interfaces/_system/system-permissions/system-permissions.vue (hunk 2 above)
Comment (paescuj) [reply app/src/interfaces/_system/system-permissions/system-permissions.vue:601]: The divider is placed between regular and system permissions.

File: app/src/interfaces/_system/system-permissions/system-permissions.vue [hunk 3]
```diff
@@ -570,117 +569,120 @@ function useGroupedPermissions() {
...
+								? appAccessMinimalPermissions.filter(
+										(permission) => permission.collection === group.collection.collection,
+								  )
+								: undefined
+						"
+						@edit-item="editItem(group.collection.collection, $event)"
+						@remove-row="removeCollection(group.collection.collection)"
+						@set-full-access-all="setFullAccessAll(group.collection.collection)"
+						@set-no-access-all="setNoAccessAll(group.collection.collection)"
+						@set-full-access="setFullAccess(group.collection.collection, $event)"
```
Comment (hanneskuettner) [inline app/src/interfaces/_system/system-permissions/system-permissions.vue:627]: > If we are doing <tfoot>, what do you think about moving this row into it as well, since it always is the last row (before the add button) and does not have the same semantic meaning as the body does? 

I meant this with my previous comment, jumped to the wrong line :D

Comment (paescuj) [reply app/src/interfaces/_system/system-permissions/system-permissions.vue:627]: You mean the system permission rows?

Comment (paescuj) [reply app/src/interfaces/_system/system-permissions/system-permissions.vue:627]: I'm afraid I don't quite understand 😢 The row for resetting app permissions is already part of the footer. Do you mean something else?

Comment (hanneskuettner) [reply app/src/interfaces/_system/system-permissions/system-permissions.vue:627]: Scratch everything I just said. I misread the diff, sorry about that!

Comment (paescuj) [reply app/src/interfaces/_system/system-permissions/system-permissions.vue:627]: No worries 👌

**Review by hanneskuettner — APPROVED** [review]
LGTM 🎸
